saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/authentication/AbstractSaml2AuthenticationRequest.java

```diff
@@ -108,7 +125,15 @@ public String getAuthenticationRequestUri() {
 
 		String relayState;
 
-		protected Builder() {
```

Comment:
We cannot remove this constructor since that could break applications that are extending this class. Instead, please mark it as `@Deprecated` so that it can be removed in a future major release.